### DISCUSSION TITLE
FCS Errors on new Intel I219-LM / SSL fails

### DISCUSSION THREAD

**Discussion by hackdroot** [ipxe/ipxe#357] [Q&A] [answered]
Recently I deployed two new workstation models which host the new Intel I219-LM [8086:0d4c], the HP Z2 G5 and HP EliteDesk 800 G6. We are seeing the following problems going through our firewall from the client to the PXE server using UEFI (legacy mode not available):

When chain loading to our SSL enabled server, the SSL traffic on the server side from the client is littered with FCS errors [ETHERNET FRAME CHECK SEQUENCE INCORRECT] and the handshake halts right before the client is set to send the key sequence. We are seeing no errors or dropped packets on the firewall however. Just a timeout from the client and server logs. We do not see these FCS errors when the kernel driver is in play on an installed system. 

This particular network card was also problematic in regards to CentOS 7, as the older e1000e driver did not work. The kmod-e1000e driver was required, or the latest 3.8.4 driver from intel.

System firmware is up to date. We installed an older intel PCI NIC, which did not display this behavior..

If the client/server are on the same network (no firewall), we still see the FCS errors, but the SSL handhshake completes.

I'd love to blame the firewall, but with no errors or dropped packets in the log, it's tough to pin down why the SSL traffic stops.

**Comment by mcb30** [comment]
> If the client/server are on the same network (no firewall), we still see the FCS errors, but the SSL handshake completes.

What is reporting the FCS errors?

It's exceptionally unlikely that the FCS errors are genuinely appearing on the wire: if the frames had checksum errors then they would be dropped by the receiver and you would never see a completed SSL handshake.  This leads me to suspect that the `ETHERNET FRAME CHECK SEQUENCE INCORRECT` error is an artifact of your packet capture process (e.g. due to segmentation offload on the receiving device) rather than an indication of an actual error in the packets.

Also: assuming that your firewall is a standard IP-layer device (rather than something that transparently intercepts Ethernet frames) then the FCS on the server side of the firewall is, in any case, constructed by the firewall's NIC and is unrelated to whatever FCS was present on the client side of the firewall.

I would suggest that you validate your packet capture process before proceeding any further with debugging.  If you do get to the point that debugging the iPXE side becomes meaningful, then building with `DEBUG=tls,x509,validator,ocsp` will show you the progress of the SSL handshake from iPXE's perspective.

**Comment by NiKiZe** [comment, marked as answer]
8086:0d4c is a (11) I219-LM, Intel releases mostly the same NIC, but just with different numbers (and pciids).
You did not include any information about which binary you are using, or the output of ifstat, but in this case you are likely using the NII efi driver. this is unlikely to be any issue at all, but if it boils down to being the firmware implementation of the firmware driver, then you could try to use ipxe.efi with #343 included, which adds native support for "version" 10-15 of i219

**Reply by mcb30** [reply]
@NiKiZe Very good point!  I have now merged #343; thanks for the reminder.

@hackdroot What output do you get from [`ifstat`](https://ipxe.org/cmd/ifstat)?

**Reply by hackdroot** [reply]
Sorry for the delay. It takes a bit of time to get a PR approved for this type of change on our network. The latest build I was able to test has resolved the issue for the desktops. The FCS errors are also no longer present on the server capture. Getting a wire capture was like pulling teeth, so I put efforts into the PR instead.

**Reply by mcb30** [reply]
@hackdroot Thanks for letting us know - glad that everything works for you now!
